x-pack/plugins/endpoint/public/embeddables/resolver/view/edge_line.tsx

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import styled from 'styled-components';
+import { useSelector } from 'react-redux';
+import { applyMatrix3 } from '../lib/vector2';
+import { Vector2 } from '../types';
+import * as selectors from '../store/selectors';
+
+export const EdgeLine = styled(
+  React.memo(
+    ({
+      className,
+      startPosition,
+      endPosition,
+    }: {
+      className?: string;
+      startPosition: Vector2;
+      endPosition: Vector2;
+    }) => {
+      const projectionMatrix = useSelector(selectors.projectionMatrix);
+      const [left, top] = applyMatrix3(startPosition, projectionMatrix);
+      const length = distance(startPosition[0], startPosition[1], endPosition[0], endPosition[1]);
```

Comment:
the readability of this stanza might be improved if Vector2 had a `distance` and an `angle` method.

```
import { angle, distance } from '../lib/vector2'
// ...
const length = distance(startPosition, endPosition)
const angle = angle(startPosition, endPosition)
// ...
```